src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp

```diff
@@ -1013,7 +1013,13 @@ void LIR_Assembler::mem2reg(LIR_Opr src, LIR_Opr dest, BasicType type, LIR_Patch
     if (UseCompressedOops && !wide) {
       __ decode_heap_oop(dest->as_register());
     }
-    __ verify_oop(dest->as_register());
+
+#if !INCLUDE_ZGC
+    if (!UseZGC) {
+      // Load barrier has not yet been applied, so ZGC can't verify the oop here
+      __ verify_oop(dest->as_register());
+    }
+#endif
```

Comment:
#if
#else
 to guarantee the isolation?

Reply:
thanks.

it's essentially add `if (!UseZGC) { .. }` before and after `__ verify_oop`. therefore, I modify it in the way below.

```
#if INCLUDE_ZGC
    if (!UseZGC) {
      // Load barrier has not yet been applied, so ZGC can't verify the oop here
#endif
      __ verify_oop(dest->as_register());
#if INCLUDE_ZGC
    }
#endif
```